src/blockdata/constants.rs

```diff
@@ -129,18 +149,24 @@ pub fn genesis_block(network: Network) -> Block {
             }
         }
         Network::Signet => {
-            // TODO: somehow obtain the block script and generate coinbase tx from it
-            let txdata = vec![bitcoin_genesis_tx()];
+            let block_script = script::Builder::new().push_opcode(opcodes::all::OP_PUSHNUM_1)
+                                                     .push_slice(&Vec::from_hex("03ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be430").unwrap())
```

Comment:
So this is the default signet signer key?

Reply:
Yes. It may change at some point, but this shouldn't be a common occurrence.